src/pages/TermsOfService.js

```diff
@@ -37,23 +44,42 @@ const TermsOfServicePage = () => {
     try {
       setBusy(true)
       const { enabled } = await Ajax().User.acceptTos()
-      termsOfService.userAcceptedTos = true
-      termsOfService.userAcceptedVersion = termsOfService.currentVersion
-      termsOfService.userNeedsToAcceptTos = false
-      authStore.update(state => ({ ...state, termsOfService }))
+
       if (enabled) {
+        const newTermsOfService = {
+          ...termsOfService,
+          userCanUseTerra: true,
+          showTosPopup: false,
+          userAcceptedVersion: termsOfService.currentVersion
+        }
+
         const registrationStatus = userStatus.registeredWithTos
-        authStore.update(state => ({ ...state, registrationStatus }))
-      }
-      if (isGracePeriod) {
-        Nav.history.goBack()
+        authStore.update(state => ({ ...state, registrationStatus, termsOfService: newTermsOfService }))
+        Nav.goToPath('root')
+      } else {
+        reportError('Error accepting TOS, unexpected backend error occurred.')
       }
     } catch (error) {
       reportError('Error accepting TOS', error)
       setBusy(false)
     }
   }
 
+  const continueButton = () => {
```

Comment:
Nit: suggest using `onAction` convention for functions that will be called on some user interaction. Suggest `onContinue` or `onClickContinue`, etc. here.